lib/providers/update_provider.dart

```diff
@@ -0,0 +1,372 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import 'dart:io';
+
+import 'package:bluecherry_client/utils/constants.dart';
+import 'package:bluecherry_client/utils/storage.dart';
+import 'package:bluecherry_client/widgets/misc.dart';
+import 'package:dio/dio.dart';
+import 'package:flutter/foundation.dart';
+import 'package:http/http.dart' as http;
+import 'package:package_info_plus/package_info_plus.dart';
+import 'package:path/path.dart' as path;
+import 'package:path_provider/path_provider.dart';
+import 'package:version/version.dart';
+import 'package:window_manager/window_manager.dart';
+import 'package:xml/xml.dart';
+
+class UpdateVersion {
+  final String version;
+  final String description;
+  final String publishedAt;
+
+  const UpdateVersion({
+    required this.version,
+    required this.description,
+    required this.publishedAt,
+  });
+
+  @override
+  String toString() =>
+      'UpdateVersion(version: $version, description: $description, publishedAt: $publishedAt)';
+
+  @override
+  bool operator ==(Object other) {
+    if (identical(this, other)) return true;
+
+    return other is UpdateVersion &&
+        other.version == version &&
+        other.description == description &&
+        other.publishedAt == publishedAt;
+  }
+
+  @override
+  int get hashCode =>
+      version.hashCode ^ description.hashCode ^ publishedAt.hashCode;
+
+  static const titleField = 'title';
+  static const descriptionField = 'description';
+  static const publishedAtField = 'pubDate';
+
+  static const rpm = 'rpm';
+  static const deb = 'deb';
+  static const tarball = 'tar.gz';
+  static const appImage = 'appimage';
+  static const linuxDownloadFileName = 'bluecherry-linux-x86_64';
+  static const windowsDownloadFileName = 'bluecherry-windows-setup';
+}
+
+class UpdateManager extends ChangeNotifier {
+  UpdateManager._();
+
+  /// `late` initialized [UpdateManager] instance.
+  static late final UpdateManager instance;
+  late final PackageInfo packageInfo;
+  late String tempDir;
+
+  /// The URL to the appcast file.
+  static const appCastUrl =
+      'https://raw.githubusercontent.com/bdlukaa/unity/upgrade/bluecherry_appcast.xml';
+
+  /// Initializes the [UpdateManager] instance & fetches state from `async`
+  /// `package:hive` method-calls. Called before [runApp].
+  static Future<UpdateManager> ensureInitialized() async {
+    instance = UpdateManager._();
+    await instance.initialize();
+    return instance;
+  }
+
+  /// If true, there is a new update available.
+  ///
+  /// If false, the user is up to date with the latest version.
+  bool get hasUpdateAvailable {
+    final currentVersion = packageInfo.version;
+    return Version.parse(currentVersion) !=
+        Version.parse(latestVersion.version);
```

Comment:
It is necessary to do a check if the latest version is not lower than the current version. It should throw an error